Documentation/network/bgp-control-plane/bgp-control-plane-v2.rst

```diff
@@ -0,0 +1,765 @@
+=============================== ===================
+
+Load Balancer Class
+~~~~~~~~~~~~~~~~~~~
+
+Cilium supports the `loadBalancerClass
+<https://kubernetes.io/docs/concepts/services-networking/service/#load-balancer-class>`__.
+When the load balancer class is set to ``io.cilium/bgp-control-plane`` or unspecified,
+Cilium will announce the ingress IPs of the Service. Otherwise, Cilium will not announce
+the ingress IPs of the Service.
+
+ExternalTrafficPolicy/InternalTrafficPolicy
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In the case of a load-balancer ingress IP or external IP advertisements,
+if the Service has ``externalTrafficPolicy: Cluster``, BGP Control Plane
+unconditionally advertises the IPs of the selected Service. When the
+Service has ``externalTrafficPolicy: Local``, BGP Control Plane keeps track of
+the endpoints for the service on the local node and stops advertisement when
+there's no local endpoint.
+
+Similarly, ``internalTrafficPolicy`` is considered for ``ClusterIP`` advertisements.
+
+.. note::
+
+    It is worth noting that when you configure ``service.addresses`` as ``ClusterIP``,
+    the BGP Control Plane only considers the configuration of the matching service's ``.spec.internalTrafficPolicy``
+    and ignores the configuration of ``.spec.externalTrafficPolicy``. For ``ExternalIP`` and
+    ``LoadBalancerIP``, it only considers the configuration of the service's ``.spec.externalTrafficPolicy``
+    and ignores the configuration of ``.spec.internalTrafficPolicy``.
+
+
+.. _bgp-override:
+
+BGP Configuration Override
+==========================
+
+``CiliumBGPNodeConfigOverride`` can be used to override some of the auto generated configuration
+on a per node basis.
+
+.. code-block:: yaml
+
+    apiVersion: cilium.io/v2alpha1
+    kind: CiliumBGPNodeConfigOverride
+    metadata:
+      name: bgpv2-cplane-dev-multi-homing-worker
+    spec:
+      bgpInstances:
+        - name: "instance-65000"
+          routerID: "192.168.10.1"
+          localPort: 1790
+          peers:
+            - name: "peer-65000-tor1"
+              localAddress: fd00:10:0:2::2
+            - name: "peer-65000-tor2"
+              localAddress: fd00:11:0:2::2
+
+
+.. note::
+    Name of ``CiliumBGPNodeConfigOverride`` resource must match the name of the node for which configuration is intended.
+    Similarly, names of the BGP instance and peers must match with what is defined under ``CiliumBGPClusterConfig``.
+
+    This is a per node configuration.
+
+RouterID
+--------
+
+When Cilium is running on an IPv4 single-stack or a dual-stack, the BGP Control Plane can use
+the IPv4 address assigned to the node as the BGP Router ID because Router ID is 32bit long, and
+we can rely on the uniqueness of the IPv4 address to make Router ID unique which is not the case
+for IPv6. Thus, when running in an IPv6 single-stack, or when the auto assignment of the Router ID
+is not desired, the administrator needs to manually define it.
```

Comment:
```suggestion
When Cilium runs on an IPv4 single-stack or a dual-stack, the BGP Control Plane can use
the IPv4 address assigned to the node as the BGP Router ID because the Router ID is 32 bit-long, and
we can rely on the uniqueness of the IPv4 address to make the Router ID unique, which is not the case
for IPv6. Thus, when running in an IPv6 single-stack, or when the auto assignment of the Router ID
is not desired, the administrator needs to manually define it.
```